environs/bootstrap/config.go

```diff
@@ -288,6 +303,33 @@ func NewConfig(attrs map[string]interface{}) (Config, error) {
 		config.CAPrivateKey = caKeyPem
 	}
 
+	// Try get the key from the attribute map, if it isn't present,
+	// try load it from the default location.
+	if sshServerHostKey, ok := attrs[SSHServerHostKeyKey].(string); ok {
+		config.SSHServerHostKey = sshServerHostKey
+	} else {
+		var userSpecified bool
+		var err error
+		config.SSHServerHostKey, userSpecified, err = readFileAttr(attrs, SSHServerHostKeyKey, SSHServerHostKeyKey)
+		if err != nil && (userSpecified || !os.IsNotExist(errors.Cause(err))) {
+			return Config{}, errors.Annotatef(err, "reading %q from file", SSHServerHostKeyKey)
+		}
+	}
+
+	// If the host key is not set, generate a new one.
+	if config.SSHServerHostKey == "" {
+		hostKey, err := rsa.GenerateKey(rand.Reader, 2048)
```

Comment:
We shouldn't be using RSA keys (and 2048 at that). We'd want `ed25519` right? It's not 1978 any more.

Reply:
We should use `pki/ssh` to generate the keys and use at least RSA3072 or ED25519 or one of ECDSA P384 or P521